cfpb/erap-lookup
# Add county filter

Add county filter and update filter component.

## Additions

- County filter
- County utils: `filterProgramsByCounty` and `generateCountyOptions`
- Unlisted option for tribe select

## Changes

- Update selects and related utils / data files to use arrays instead of objects

## Testing

1. Check out branch
2. To view with CFPB design system styles, paste [this gist's contents](https://gist.github.com/virginiacc/9eaa4b840599288ae48d02f4c7ed2c77) in `App.css`
3. yarn start
4. Verify that county filter works as expected

## Screenshots

### County dropdown displays when state with sufficient results is selected
<img width="808" alt="Screen Shot 2021-07-22 at 2 21 52 PM" src="https://user-images.githubusercontent.com/778171/126726358-17e12ce6-09b2-458f-88e9-79079eb419e1.png">

### County dropdown options with unlisted option at top
<img width="843" alt="Screen Shot 2021-07-22 at 2 22 03 PM" src="https://user-images.githubusercontent.com/778171/126726378-3303adf7-6929-472d-8ab8-33df9f683449.png">

### Results for county selection
<img width="815" alt="Screen Shot 2021-07-22 at 2 22 15 PM" src="https://user-images.githubusercontent.com/778171/126726380-1a76695f-7a50-4c92-9b04-b215db627c5d.png">

### Results for unlisted county selection
<img width="841" alt="Screen Shot 2021-07-22 at 2 22 27 PM" src="https://user-images.githubusercontent.com/778171/126726393-b2ea6429-d7fe-4c3c-b0b7-33b1dfe89369.png">

### All results 
<img width="816" alt="Screen Shot 2021-07-22 at 2 22 50 PM" src="https://user-images.githubusercontent.com/778171/126726397-dbd378da-0bea-41e1-86d3-e33bae7f1167.png">

### Tribe unlisted option selected
<img width="816" alt="Screen Shot 2021-07-22 at 2 23 00 PM" src="https://user-images.githubusercontent.com/778171/126726422-7aa7e02b-291d-4253-ba5b-46a641c234b3.png">
